src/StagedProposalProcessor.sol

```diff
@@ -312,33 +319,40 @@ contract StagedProposalProcessor is IProposal, PluginUUPSUpgradeable {
             return false;
         }
 
-        // Allow `stageDuration` to pass for plugins to have veto possibility.
+        // Allow `voteDuration` to pass for plugins to have veto possibility.
         if (stage.vetoThreshold > 0) {
-            if (proposal.lastStageTransition + stage.stageDuration > block.timestamp) {
+            if (proposal.lastStageTransition + stage.voteDuration > block.timestamp) {
                 return false;
             }
         }
 
         uint256 approvals = 0;
         uint256 vetoes = 0;
+        uint256 vetoesEarlyCount = 0;
         for (uint256 i = 0; i < stage.plugins.length; ) {
             Plugin storage plugin = stage.plugins[i];
             address allowedBody = plugin.allowedBody;
 
+            uint256 pluginProposalId = pluginProposalIds[_proposalId][currentStage][plugin.pluginAddress];
+            
             if (pluginResults[_proposalId][currentStage][plugin.proposalType][allowedBody]) {
                 if (plugin.proposalType == ProposalType.Approval) {
                     ++approvals;
                 } else {
                     ++vetoes;
                 }
+            } else if(stage.vetoThreshold > 0 && !plugin.isManual && pluginProposalId != 2**256 - 1) {
+                if(IProposal(stage.plugins[i].pluginAddress).canExecute(pluginProposalId)){
+                    vetoesEarlyCount++;
+                }
```

Comment:
here we are assuming that `plugin.proposalType` is `Veto` but it could also be `Approval`.

Reply:
Fixed in https://github.com/aragon/staged-proposal-processor-plugin/pull/3/commits/1bfb7a9be86fde17a4c3f537b1e0469aa57ab8cf and I removed also `vetoEarlyCounts` as I don't think it was necessary.